lib/matplotlib/widgets.py

```diff
@@ -2984,20 +3011,29 @@ class PolygonSelector(_SelectorWidget):
     ----------
     ax : `~matplotlib.axes.Axes`
         The parent axes for the widget.
+
     onselect : function
         When a polygon is completed or modified after completion,
         the *onselect* function is called and passed a list of the vertices as
         ``(xdata, ydata)`` tuples.
+
     useblit : bool, default: False
-    lineprops : dict, default: \
-``dict(color='k', linestyle='-', linewidth=2, alpha=0.5)``.
+
+    lineprops : dict
         Artist properties for the line representing the edges of the polygon.
-    markerprops : dict, default: \
-``dict(marker='o', markersize=7, mec='k', mfc='k', alpha=0.5)``.
+        Default::
+
+            dict(color='k', linestyle='-', linewidth=2, alpha=0.5)
+
+    handle_props : dict
```

Comment:
I see what you what you mean, but I am convinced that simplifying the API and making it more generic is worth it at the cost of the being slightly less precise. I agree that using a more specific terminology is more explicit but at the end of the day, when we create a selector - regardless of what selector it is - what we want is to set the properties of the patch and/or the handle, regardless of what their are or of their implementation details.
For example, as an average matplotlib user, I don't know what are the different parameters between `lineprops` and `markerprops`, I will need to look it up.

In 35ee92d, I have changed the various patch properties (`rectprops`, `lineprops`) argument to `props`, so that now we have only `props` and `handleprops` for all selectors.
To address #20618, we could add two method to set the properties of the selector patch and the handle separately. The idea is that either at initialisation or when using `set_props` or `set_handle_props`, we could set the properties regardless of what type of object they are.

I am happy to revert 35ee92d if others also think that this is going in the wrong direction!

Reply:
> For example, as an average matplotlib user, I don't know what are the different parameters between `lineprops` and `markerprops`, I will need to look it up.

Conversely, I argue that `props` is even less helpful. The different selectors are realized by different Artists and thus have different properties. We cannot hide that fact. `SpanSelector` accepts `facecolor`, `LassoSelector` accepts `solid_linestyle`, and neither accepts the other one respectively. With `props` the user has to lookup what is supported as well. But it's more surpising and harder to remember that the same parameter name accepts different inputs depending on the class.